source/components/form/form.ts

```diff
@@ -0,0 +1,73 @@
+// /// <reference path='../../../typings/node/node.d.ts' />
+
+'use strict';
+
+import * as angular from 'angular';
+import * as _ from 'lodash';
+
+import { IFormValidator } from '../../types/formValidators';
+import { IAutosaveService, IAutosaveServiceFactory, factoryName as autosaveFactoryName, moduleName as autosaveModule } from '../../services/autosave/autosave.service';
+
+export let moduleName: string = 'rl.ui.components.form';
+export let componentName: string = 'rlForm';
+export let controllerName: string = 'rlFormController';
+
+export interface IFormBindings {
+	saving: boolean;
+	save(): void;
+}
+
+export interface IFormScope extends angular.IScope {
+	rlForm: IFormValidator;
+}
+
+export class FormController implements IFormBindings {
+	saving: boolean;
+	save: { (): void };
+
+	autosave: IAutosaveService;
+
+	static $inject: string[] = ['$element', '$scope', '$timeout', '$q', autosaveFactoryName];
+	constructor(private $element: angular.IAugmentedJQuery
+			, private $scope: IFormScope
+			, private $timeout: angular.ITimeoutService
+			, private $q: angular.IQService
+			, private autosaveFactory: IAutosaveServiceFactory) { }
+
+	$onInit(): void {
+		this.$element.find('form').on('submit', (): void => {
+			this.autosave.validateAndSave();
+			this.$scope.$apply();
```

Comment:
It doesn't seem like this is going to prevent the actual form from submitting, which could result in some unexpected behavior.